src/adapters/smartContractQueriesAdapter.ts

```diff
@@ -0,0 +1,44 @@
+import { Address } from "../address";
+import { IAddress } from "../interface";
+import { IContractQueryResponse } from "../interfaceOfNetwork";
+import { SmartContractQuery, SmartContractQueryResponse } from "../smartContractQuery";
+
+interface INetworkProvider {
+    queryContract(query: ILegacyQuery): Promise<IContractQueryResponse>;
+}
+
+interface ILegacyQuery {
+    address: IAddress;
+    caller?: IAddress;
+    func: { toString(): string };
+    value?: { toString(): string };
+    getEncodedArguments(): string[];
+}
+
+export class SmartContractQueriesAdapter {
```

Comment:
Let's think of a general rule for naming the classes in the `adapters` package.

E.g., this can be, I think, `NetworkProviderAsQueryRunnerAdapter`.

And the upcoming adapter required by `TransactionWatcher` would be: `ProxyNetworkProviderAsTransactionFetcherAdapter`.

How does this sound? We should ask for some opinions.

Reply:
From what I see here the query method actually returns a SmartContractQueryResponse and it is the only method so, I would keep the SmartContractQueriesAdapter